### PR TITLE
updated documentation about draft in node.js

### PR DESCRIPTION
**Pull request by etimr**

### CODE REVIEW

File: node.js/fiori.md [hunk 1]
```diff
@@ -113,37 +36,59 @@ Lean draft is enabled by default. Add this to your `cds` configuration to disabl
   }
 }
 ```
+### Handlers Registration { #draftevents}
 
-### Differences to Previous Version
+Class `ApplicationService` provides built-in support for Fiori Draft. All CRUD events are supported for both, active and draft entities. 
 
-- Draft-enabled entities have corresponding CSN entities for drafts:
+Please note, that `CREATE` and `UPDATE` events for an active entity are not the same as in the case of a nondraft implementation. You can find more details about these events below.
```
Comment (David-Kunz) [inline node.js/fiori.md:43]: What do you mean with that? Can you provide an example?

File: node.js/fiori.md [hunk 2]
```diff
@@ -113,37 +36,59 @@ Lean draft is enabled by default. Add this to your `cds` configuration to disabl
...
     srv.on('boundActionOrFunction', 'MyEntity.drafts', /*...*/)
+  ```
+
+It is also possible to use the array variant to register a handler for both entities, for example: `srv.on('boundActionOrFunction', ['MyEntity', 'MyEntity.drafts'], /*...*/)`.
+
+Additionally, you can add your logic to the draft specific events as follows:
+
+  ```js
+    // creates new empty draft
     srv.on('NEW', 'MyEntity.drafts', /*...*/)
```
Comment (David-Kunz) [inline node.js/fiori.md:64]:
```suggestion
    // When a new draft is created
    srv.on('NEW', 'MyEntity.drafts', /*...*/)
```

File: node.js/fiori.md [hunk 3]
```diff
@@ -113,37 +36,59 @@ Lean draft is enabled by default. Add this to your `cds` configuration to disabl
...
+    // cancels draft
     srv.on('CANCEL', 'MyEntity.drafts', /*...*/)
```
Comment (David-Kunz) [inline node.js/fiori.md:66]:
```suggestion
    // When a draft is discarded
    srv.on('CANCEL', 'MyEntity.drafts', /*...*/)
```

File: node.js/fiori.md [hunk 4]
```diff
@@ -113,37 +36,59 @@ Lean draft is enabled by default. Add this to your `cds` configuration to disabl
...
+    // creates new draft from existing active entity
```
Comment (David-Kunz) [inline node.js/fiori.md:67]:
```suggestion
    // When a new draft is created from an active instance
```

File: node.js/fiori.md [hunk 5]
```diff
@@ -113,37 +36,59 @@ Lean draft is enabled by default. Add this to your `cds` configuration to disabl
...
     srv.on('EDIT', 'MyEntity', /*...*/)
     srv.on('SAVE', 'MyEntity', /*...*/)
```
Comment (David-Kunz) [inline node.js/fiori.md:69]:
```suggestion
    // When the active entity is changed
    srv.on('SAVE', 'MyEntity', /*...*/)
```

File: node.js/fiori.md [hunk 6]
```diff
@@ -113,37 +36,59 @@ Lean draft is enabled by default. Add this to your `cds` configuration to disabl
...
-    ```
+  ```
+
+- The `CANCEL` event is triggered when you cancel the draft. In this case, the draft entity is deleted and the active entity is not changed.
+- The `EDIT` event is triggered when you start editing an active entity. As a result `MyEntity.drafts` is created. 
+- The `SAVE` event is a shortcut for `['UPDATE', 'CREATE']` on an active entity. This event is triggered when you press `SAVE` button in UI after finishing editing your draft. Please note, that composition children of the active entity will also be updated or created. The `SAVE` event also means that `MyEntity.drafts` is deleted and all changes are written into `MyEntity`.
```
Comment (David-Kunz) [inline node.js/fiori.md:74]:
```suggestion
- The `SAVE` event is just a shortcut for `['UPDATE', 'CREATE']`. This event is also triggered when you press the `SAVE` button in UI after finishing editing your draft. Please note, that composition children of the active entity will also be updated or created.
```

> Please note, that composition children of the active entity will also be updated or created.

Not sure why this is mentioned, is this important?

File: node.js/fiori.md [hunk 7]
```diff
@@ -113,37 +36,59 @@ Lean draft is enabled by default. Add this to your `cds` configuration to disabl
...
+- Draft-enabled entities have corresponding CSN entities for drafts:
+
+    ```js
+    const { MyEntity } = srv.entities
+    MyEntity.drafts // points to model.definitions['MyEntity.drafts']
+    ```
 
     ::: info Compatibility flag
     For compatibility to previous variants, set `cds.fiori.draft_compat` to `true`.
     :::
```
Comment (David-Kunz) [inline node.js/fiori.md:87]: That should be below the handler registration section.

File: node.js/fiori.md [hunk 8]
```diff
@@ -160,3 +105,12 @@ Lean draft is enabled by default. Add this to your `cds` configuration to disabl
 
 - Draft-related properties (with the exception of `IsActiveEntity`) are only computed for the target entity, not for expanded sub entities since this is not required by Fiori Elements.
 - Manual filtering on draft-related properties is not allowed, only certain draft scenarios are supported.
+
+
+## Draft Locks
```
Comment (David-Kunz) [inline node.js/fiori.md:110]: Maybe move that section up so that `Difference to Previous Version` is at the bottom.

File: node.js/fiori.md (hunk 1 above)
Comment (etimr) [reply node.js/fiori.md:43]: I remember issues, where it wasn't clear for stakeholders, that direct actions on an active entity are not possible, for example CREATE on MyEntity will not be triggered if no draft exists. I can remove it, if the description is so unclear

File: node.js/fiori.md (hunk 6 above)
Comment (etimr) [reply node.js/fiori.md:75]: I wanted to explain a life cycle of a draft entity. `SAVE` event is registered on an active entity, so maybe it's not completely transparent that draft is deleted afterwards and for example in an `srv.after('SAVE', 'MyEntity', /*...*/)` one can't access  `MyEntity.drafts` anymore.

**Review by David-Kunz — APPROVED** [review]

**Review by smahati — APPROVED** [review]

**Pull request merged**